gui/ports.js

```diff
@@ -80,9 +80,23 @@ elmectron.ports.chooseFolder.subscribe(() => {
   ipcRenderer.send('choose-folder')
 })
 
-ipcRenderer.on('synchronization', (event, url, deviceName) => {
-  elmectron.ports.synchonization.send([url, deviceName])
-})
+ipcRenderer.on(
+  'sync-config',
+  (event, address, deviceName, deviceId, capabilities, flags) => {
+    const partialSyncEnabled =
+      flags['settings.partial-desktop-sync.show-synced-folders-selection']
+    elmectron.ports.syncConfig.send({
+      address,
+      deviceName,
+      deviceId,
+      capabilities,
+      flags: {
+        partialSyncEnabled:
+          partialSyncEnabled != null ? partialSyncEnabled : true
```

Comment:
`true` as the default value?

Reply:
Yes. The idea is to be able to remove the flag later without disabling the feature. The corrolary is that we'll have to set the flag for every context before releasing this code to make sure it's not enable by default.

Reply:
Good luck setting the flag for the self-hosted ;-)

Maybe you should start with `false`, and change the default value in a future release.

Reply:
Or I make sure the Settings app has the same behavior so they get selective synchronization by default?!

Reply:
Yes, but you need to release settings and cozy-desktop at the same time, which is what you wanted to avoid in the first place, no?

Reply:
Yes, that's why I want to use the flag. I don't see how I can avoid the release issue for self hosted users though. Do you have any idea?

Reply:
Maybe we could use the capabilities for that?

Reply:
Actually, I can see a way that's actually gonna be needed: define the minimum Desktop version for the Settings to show the configuration button.
But that means knowing in advance which version of the Desktop will be able to do selective synchronization.

Reply:
1. We can make a release of cozy-desktop, with `false` as the default value here (version `x.y.z`)
2. We release settings, with partial sync enabled if at least one of those conditions is true:
    - the flag is activated
    - the cozy-desktop client is at least on version `x.y.z+1`
3. We make a new release of cozy-desktop that just swaps the default value to `true` (version `x.y.z+1`).

What do you think?

Reply:
That works only if we don't have to make an intermediate Desktop release without partial sync. That's why setting the expected version in advance is not fool proof.

Reply:
I'm gonna set the default value to `false` until we figure out the right way to plan the release.